docs/DeveloperGuide.md

```diff
@@ -3,36 +3,144 @@
+|v2.0|user|Export my financial reports as a CSV|I can share my expenditure with my peers|
+|v2.0|user|Visualise my transactions in more meaningful diagrams|I can better understand my spending|
 
 ## Non-Functional Requirements
 
-{Give non-functional requirements}
+1. **Security**: User authentication must be secure, with a limit on login attempts to prevent unauthorized access.
+2. **Usability**: The application should be easy to use, with clear instructions and feedback for users.
+3. **Performance**: The application should respond quickly to user inputs, with minimal delays in processing transactions.
+4. **Reliability**: Data storage should be reliable, ensuring that user data is not lost between sessions.
+
 
 ## Glossary
 
-* *glossary item* - Definition
+1. **Inflow** - A financial transaction representing income or money received.
+2. **Outflow** - A financial transaction representing expenses or money spent.
+
 
 ## Instructions for manual testing
 
-{Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing}
+### Login Procedure:
+- Start the application and enter your username and password when prompted.
+- Test incorrect passwords to ensure the application correctly limits login attempts. 
+- [v2.0] If there are three incorrect attempts, the application will automatically exit and the user has to try again.
+
+### Adding Transactions:
+- Use the `add-inflow` and `add-outflow` commands to add new transactions, following the command format provided in the `help` command.
+- Attempt to add transactions with missing or incorrect information to test validation.
+
+### Editing and Deleting Transactions:
+- Use the `edit-inflow` and `edit-outflow` commands to modify existing transactions.
+- Use the `delete-inflow` and `delete-outflow` commands to remove transactions.
+- Try editing or deleting transactions that do not exist to test error handling.
+
+### Viewing Transaction History:
+- Use the `view-history` command to display recent transactions.
+- Test with different numbers of transactions to view.
+- [v2.0] Additional bar charts will be shown to display the different percentage of categories used in the past n transactions.
+
+### Inactivity Timeout:
+- After logging in, do not input any commands for the duration specified by the inactivity timer to test automatic logout.
+- A maximum of 2.5 minutes of inactivity will trigger the application to check with the user whether the user is still active or not. Input `yes` to indicate continued activity, otherwise input `no` to exit. 
+- 30 seconds after the 2.5 min (3 min), the application will automatically exit.
+- Implemented to provide enhanced security, so no third party can access finance transactions.
+
+
+![photo_2024-04-04 23 54 55](https://github.com/AY2324S2-CS2113-F14-4/tp/assets/88567790/aa2df04e-c493-443e-a4f1-e3cfc506e56a)
```

Comment:
Perhaps you could provide a header for the diagrams / include them near the sequence they are describing so that the formatting of the DG is better instead of putting them at the end of the document.